### PR TITLE
Feature/retrieval market provider query

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Add retrieval market provider query handler.

### CODE REVIEW

File: core/markets/retrieval/provider/impl/retrieval_provider_impl.cpp [hunk 1]
```diff
@@ -3,23 +3,30 @@
...
+        piece_storage_{std::move(piece_storage)},
+        config_{} {}
 
   void RetrievalProviderImpl::start() {
-    host_service_->setProtocolHandler(
-        kQueryProtocolId, [self = shared_from_this()](StreamShPtr stream) {
-          self->query_responder_->onNewRequest(
-              std::make_shared<CborStream>(stream));
-        });
+    // TODO move set handler to init
```
Comment (artyom-yurin) [inline core/markets/retrieval/provider/impl/retrieval_provider_impl.cpp:24]: Please correlate with the task in Jira

File: core/markets/retrieval/provider/retrieval_provider_types.hpp [hunk 2]
```diff
@@ -24,101 +20,6 @@ namespace fc::markets::retrieval::provider {
     uint64_t interval_increase;
```
Comment (artyom-yurin) [inline core/markets/retrieval/provider/retrieval_provider_types.hpp:20]: Maybe move structure to primitives

File: core/storage/piece/impl/piece_storage_error.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#define CPP_FILECOIN_CORE_STORAGE_PIECE_IMPL_PIECE_STORAGE_ERROR_HPP
+
+#include "common/outcome.hpp"
+
+namespace fc::storage::piece {
+
+  /**
+   * @enum Piece storage errors
+   */
+  enum class PieceStorageError { kPieceNotFound, kPayloadNotFound };
```
Comment (artyom-yurin) [inline core/storage/piece/impl/piece_storage_error.hpp:16]:
```suggestion
  enum class PieceStorageError { kPieceNotFound = 1, kPayloadNotFound };
```

File: core/markets/retrieval/provider/impl/retrieval_provider_impl.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/markets/retrieval/provider/impl/retrieval_provider_impl.cpp]: todo removed

File: core/markets/retrieval/provider/retrieval_provider_types.hpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply core/markets/retrieval/provider/retrieval_provider_types.hpp]: File removed, struct moved to provider.

File: core/storage/piece/impl/piece_storage_error.hpp (hunk 3 above)
Comment (Alexey-N-Chernyshov) [reply core/storage/piece/impl/piece_storage_error.hpp]: fix applied

**Pull request merged**